main.cc

```diff
@@ -2111,9 +2111,13 @@ sharded<locator::shared_token_metadata> token_metadata;
             }).get();
 
             checkpoint(stop_signal, "join cluster");
+            // Allow abort during join_cluster since bootstrap or replace
+            // can take a long time.
+            stop_signal.ready(true);
             with_scheduling_group(maintenance_scheduling_group, [&] {
                 return ss.local().join_cluster(proxy, service::start_hint_manager::yes, generation_number);
             }).get();
+            stop_signal.ready(false);
```

Comment:
If it's ready, then it's ready, why disable immediate handing of stop signal after join?

Reply:
> why disable immediate handing of stop signal after join?

just to simplify life for the rest of the startup code